src/elexmodel/handlers/data/VersionedData.py

```diff
@@ -143,15 +147,18 @@ def compute_estimated_margin(df):
             # Compute batch_margin using NumPy
             # this is the difference in dem_votes - the difference in gop_votes divided by the difference in total votes
             # that is, this is the normalized margin in the batch of votes recorded between versions
-            batch_margin = (
-                np.diff(results_dem, append=results_dem[-1]) - np.diff(results_gop, append=results_gop[-1])
-            ) / np.diff(results_weights, append=results_weights[-1])
+            with warnings.catch_warnings():
```

Comment:
I vaguely remember us doing something previously to avoid this already. At least stopping it turning up in the logs. Am I misremembering?

Reply:
Interesting 🤔  Maybe it was another spot where we can occasionally get divisions by zero? Without this, right now I get warnings here when running the unit tests 🤔  Curious 🤔

Reply:
If you re-run the 2024 in the testbed (without writing anything to s3), do you see this warning?

Reply:
Alright so I ran this:

```
python run.py 2024-11-05_USA_G redo --office_id S --estimands="['margin']" --features "['baseline_normalized_margin']" --end_timestamp 2024-11-06T12:00:00-05:00 --model_parameters='{"extrapolation" : True, "versioned_start_date" : "2024-11-05T21:00:00-05:00", "versioned_end_date" : "2024-11-05T21:30:00-05:00"}
```

And the answer right now is no, BUT, I noticed that even though we're capturing warnings, they're not currently coming through in the logs.  Having changed that, this actually generates a lot of warnings 😬  

PR incoming 👀

Reply:
Hi @jjcherian ! Hope you're doing well 😄 👋🏻 

We noticed that these few lines can generate quite a few `invalid value encountered in divide` and `divide by zero encountered in divide` warnings.  If you clone the `develop` branch and run, for example,

```
elexmodel 2024-11-05_USA_G --estimands=margin --features=baseline_normalized_margin --office_id=S --geographic_unit_type=county --pi_method bootstrap --national_summary --model_parameters '{"extrapolation" : True, "versioned_start_date" : "2024-11-05T21:00:00-05:00", "versioned_end_date" : "2024-11-05T21:30:00-05:00"}'
```

you should see them.  We were wondering what the best way to handle these warning is. Right now, in this PR, I've added code to ignore (and therefore silence) them, but do you think there's a better way to resolve these? 🤔 

Thanks in advance 😄